gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/DagActionStoreChangeMonitor.java

```diff
@@ -313,10 +300,28 @@ protected void createMetrics() {
     this.nullDagActionTypeMessagesMeter = this.getMetricContext().contextAwareMeter(RuntimeMetrics.GOBBLIN_DAG_ACTION_STORE_MONITOR_NULL_DAG_ACTION_TYPE_MESSAGES);
     this.produceToConsumeDelayMillis = this.getMetricContext().newContextAwareGauge(RuntimeMetrics.GOBBLIN_DAG_ACTION_STORE_PRODUCE_TO_CONSUME_DELAY_MILLIS, () -> produceToConsumeDelayValue);
     this.getMetricContext().register(this.produceToConsumeDelayMillis);
+
+    // Setup proxy for launch submission metrics
+    this.ON_STARTUP = new LaunchSubmissionMetricProxy(this.successfulLaunchSubmissionsOnStartup, this.failedLaunchSubmissionsOnStartup);
+    this.POST_STARTUP = new LaunchSubmissionMetricProxy(this.successfulLaunchSubmissions, this.failedLaunchSubmissions);
```

Comment:
putting the initialization here means NPE during any invocation of `submitFlowToDagManagerHelper` that may come prior.

if you can't rework, I suggest temporarily initializing these according to the Null Pattern:
```
LaunchSubmissionMetricProxy ON_STARTUP = new LaunchSubmissionMetricProxy {
  @Override public void markSuccess() { }
  @Override public void markFailure() { }
}
```

Reply:
the other alternative is to use the Null/NoOp Pattern in the call site, e.g.:
```
protected void submitFlow(DagAction dagAction, boolean isStartup) {
  submitFlow(dagAction, (isStartup ? ON_STARTUP : POST_STARTUP).orElse(new LaunchSubmissionMetricProxy { /* as in above comment */}));
}
```
i.e. make these two `volatile Optional<LSMP>`